src/dhcp_device.cpp

```diff
@@ -288,7 +292,14 @@ static void client_packet_handler(dhcp_device_context_t *context, uint8_t *buffe
                     ntohs(udp->len) : buffer_sz - UDP_START_OFFSET - sizeof(struct udphdr);
         int dhcp_option_sz = dhcp_sz - DHCP_OPTIONS_HEADER_SIZE;
         const u_char *dhcp_option = buffer + dhcp_option_offset;
-        
+        uint32_t magic_cookie = dhcphdr[MAGIC_COOKIE_OFFSET] << 24 | dhcphdr[MAGIC_COOKIE_OFFSET + 1] << 16 |
```

Comment:
How can we ensure this OFFSET not overflow?

Reply:
this line, it would require packet have enough length to contain magic cookie
![image](https://github.com/user-attachments/assets/14644f8a-8f29-4ead-a4d4-b1a0c9394778)